update EWSO365 readme (#35012)
demisto/content: Packs/MicrosoftExchangeOnline/Integrations/EWSO365/README.md

Exchange Web Services (EWS) provides the functionality to enable client applications to communicate with the Exchange server. EWS provides access to much of the same data that is made available through Microsoft Office Outlook.

The EWS O365 integration implants EWS leading services. The integration allows getting information on emails and activities in a target mailbox, and some active operations on the mailbox such as deleting emails and attachments or moving emails from folder to folder.

## Use Cases

The EWS integration can be used for the following use cases.

* Monitor a specific email account and create incidents from incoming emails to the defined folder.  
    Follow the instructions in the [Fetched Incidents Data section](https://xsoar.pan.dev/docs/reference/integrations/ewso365#fetch-data).

* Search for an email message across mailboxes and folders.  

>     This can be achieved in the following ways:

Reviewer: ```suggestion

```

>     1. Use the `ews-search-mailbox` command to search for all emails in a specific folder within the target mailbox.  

Reviewer: ```suggestion
    Use the `ews-search-mailbox` command to search for all emails in a specific folder within the target mailbox.  
```

>         Use the query argument to narrow the search for emails sent from a specific account and more.

Reviewer: ```suggestion
     Use the query argument to narrow the search for emails sent from a specific account and more.
```

>     2. This command retrieves the _ItemID_ field for each email item listed in the results. The `ItemID` value can be used in the `ews-get-items` command in order to get more information about the email item itself.

Reviewer: ```suggestion
    This command retrieves the _ItemID_ field for each email item listed in the results. The `ItemID` value can be used in the `ews-get-items` command in order to get more information about the email item itself.
```

* Get email attachment information.  
    Use the `ews-get-attachment` command to retrieve information on one attachment or all attachments of a message at once. It supports both file attachments and item attachments (e.g., email messages).

* Delete email items from a mailbox.  
    First, make sure you obtain the email item ID. The item ID can be obtained with one of the integration’s search commands.  
    Use the `ews-delete-items`<span> command </span>to delete one or more items from the target mailbox in a single action.  
    A less common use case is to remove emails that were marked as malicious from a user’s mailbox.  
    You can delete the items permanently (hard delete) or delete the items (soft delete), so they can be recovered by running the `ews-recover-messages` command.

## Architecture

> This integration is based on the `exchangelib` python module. For more information about the module check the [documentation](https://ecederstrand.github.io/exchangelib/).

Reviewer: ```suggestion
This integration is based on the `exchangelib` python module. For more information about the module, check the [documentation](https://ecederstrand.github.io/exchangelib/).
```

## Set up the Third Party System

There are two application authentication methods available.
Follow your preferred method's guide on how to use the admin consent flow in order to receive your authentication information:

* [Cortex XSOAR Application](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#cortex-xsoar-application)
* [Self-Deployed Application](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#self-deployed-application) - Client Credential Flow.

For more details about the authentication used in this integration, see [Microsoft Integrations - Authentication](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication).

## Permissions

In order to function as expected, the service account should have:

> **Impersonation rights** - In order to perform actions on the target mailbox of other users, the _service account_ must be part of the `ApplicationImpersonation` role. For more information and guide on how to set up the permission, follow the [Microsoft Documentation](https://learn.microsoft.com/en-us/exchange/client-developer/exchange-web-services/impersonation-and-ews-in-exchange).

Reviewer: ```suggestion
**Impersonation rights** - In order to perform actions on the target mailbox of other users, the _service account_ must be part of the `ApplicationImpersonation` role. For more information and instructions on how to set up the permission, see [Microsoft Documentation](https://learn.microsoft.com/en-us/exchange/client-developer/exchange-web-services/impersonation-and-ews-in-exchange).
```

> Most command require this permission to function correctly. This permission is specified in each relevant command's Permission section. For more information follow the [Microsoft Documentation](https://learn.microsoft.com/en-us/exchange/client-developer/exchange-web-services/impersonation-and-ews-in-exchange). 

Reviewer: ```suggestion
Most command require this permission to function correctly. This permission is specified in each relevant command's Permission section. For more information, see [Microsoft Documentation](https://learn.microsoft.com/en-us/exchange/client-developer/exchange-web-services/impersonation-and-ews-in-exchange). 
```

> **eDiscovery** permissions to the Exchange Server. For Users to be able to use Exchange Server In-Place eDiscovery, they must be added to the Discovery Management role group. Members of the Discovery Management role group have Full Access mailbox permissions to the default discovery mailbox, which is called Discovery Search Mailbox, including access to sensitive message content. For more information see the [Microsoft documentation](https://technet.microsoft.com/en-us/library/dd298059(v=exchg.160).aspx).

Reviewer: ```suggestion
**eDiscovery** permissions to the Exchange Server. For users to be able to use Exchange Server In-Place eDiscovery, they must be added to the Discovery Management role group. Members of the Discovery Management role group have Full Access mailbox permissions to the default discovery mailbox, which is called Discovery Search Mailbox, including access to sensitive message content. For more information, see the [Microsoft documentation](https://technet.microsoft.com/en-us/library/dd298059(v=exchg.160).aspx).
```

The need for this permission is specified in each relevant command's Permission section.

**full_access_as_app** - The _application used for authentication_ requires this permission to gain access to the Exchange Web Services.
To set this permission follow these steps:

1. Navigate to **Home** > **App registrations**.

> 2. Search for your app under ‘all applications’.

Reviewer: ```suggestion
2. Search for your app under *all applications*.
```

3. Click **API permissions** > **Add permission**.
4. Search for `Office 365 Exchange Online` API > `Application Permission`> `full_access_as_app` permission. 

> For more information on this permission, visit [the Microsoft documentation](https://docs.microsoft.com/en-us/exchange/client-developer/exchange-web-services/how-to-authenticate-an-ews-application-by-using-oauth#configure-for-app-only-authentication).

Reviewer: ```suggestion
For more information on this permission, see [the Microsoft documentation](https://docs.microsoft.com/en-us/exchange/client-developer/exchange-web-services/how-to-authenticate-an-ews-application-by-using-oauth#configure-for-app-only-authentication).
```

| UPN Address | When provided, the target mailbox if it's different from the Email Address. Otherwise, the Email Address is used. | False |
| Name of the folder from which to fetch incidents | Supports Exchange Folder ID and sub-folders, e.g., Inbox/Phishing. | True |
| Access Type | Run the commands using `Delegate` or `Impersonation` access types. | False |
| Public Folder |  Whether the folder to be fetched from is public. Public folders can store and organize emails on specific topics or projects. Public folders are usually listed under the "Public Folders" section in the navigation pane in the product itself. | False |
| Fetch incidents |  | False |
| Incident type |  | False |
| First fetch timestamp (&lt;number&gt; &lt;time unit&gt;, e.g., 12 hours, 7 days) |  | False |
| Maximum number of incidents per fetch (up to 200). Performance might be affected by a value higher than 50. |  | False |

> | Mark fetched emails as read | Mark fetched emails as read. | False |

Reviewer: The description does not add anything since it is just a repeat of the parameter. I recommend just remove the description.

> | Mark fetched emails as read | Mark fetched emails as read. | False |

Author: ```suggestion
| Mark fetched emails as read | | False |
```

| Timeout (in seconds) for HTTP requests to Exchange Server |  | False |
| Trust any certificate (not secure) |  | False |
| Use system proxy settings |  | False |
| Run as a separate process (protects against memory depletion) |  | False |

> | Use a self deployed Azure Application | Select this checkbox if you are using a self-deployed Azure application. | False |

Reviewer: ```suggestion
| Use a self-deployed Azure Application | Select this checkbox if you are using a self-deployed Azure application. | False |
```

    First, make sure you obtain the email item ID. The item ID can be obtained with one of the integration’s search commands.  
    Use the `ews-delete-items`<span> command </span>to delete one or more items from the target mailbox in a single action.  
    A less common use case is to remove emails that were marked as malicious from a user’s mailbox.  
    You can delete the items permanently (hard delete) or delete the items (soft delete), so they can be recovered by running the `ews-recover-messages` command.

## Architecture

This integration is based on the `exchangelib` python module. For more information about the module check the [documentation](https://ecederstrand.github.io/exchangelib/).

## Special Notes/Disclaimers

No known disclaimers.

## Known Limitations

No known limitations.

## System Integration Setup (Device side steps)

There are two application authentication methods available.
Follow your preferred method's guide on how to use the admin consent flow in order to receive your authentication information:

* [Cortex XSOAR Application](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#cortex-xsoar-application)

> * [Self-Deployed Application](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#self-deployed-application)

Reviewer 2: with self-deployed only the client-creds flow is supported, maybe change to:
```suggestion
* [Self-Deployed Application](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#self-deployed-application) - Client Credentials Flow
```

For more details about the authentication used in this integration, see [Microsoft Integrations - Authentication](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication).

## Permissions

> **full_access_as_app** - To set this permission follow [the Microsoft documentation](https://docs.microsoft.com/en-us/exchange/client-developer/exchange-web-services/how-to-authenticate-an-ews-application-by-using-oauth#configure-for-app-only-authentication).

Reviewer 2: Let's discuss with Shirley the best way to document how to set the permissions.

> You can't manage the **Office 365 Exchange Online** app permissions via the Azure portal.

Reviewer 2: Maybe start a new line before
![Uploading image.png…]()

| Access Type | Run the commands using `Delegate` or `Impersonation` access types. | False |
| Public Folder |  Whether the folder to be fetched from is public. Public folders can store and organize emails on specific topics or projects. Public folders are usually listed under the "Public Folders" section in the navigation pane in the product itself. | False |
| Fetch incidents |  | False |
| Incident type |  | False |
| First fetch timestamp (&lt;number&gt; &lt;time unit&gt;, e.g., 12 hours, 7 days) |  | False |
| Maximum number of incidents per fetch (up to 200). Performance might be affected by a value higher than 50. |  | False |
| Mark fetched emails as read | Mark fetched emails as read. | False |
| Timeout (in seconds) for HTTP requests to Exchange Server |  | False |
| Trust any certificate (not secure) |  | False |
| Use system proxy settings |  | False |
| Run as a separate process (protects against memory depletion) |  | False |
| Use a self deployed Azure Application | Select this checkbox if you are using a self-deployed Azure application. | False |
| Incidents Fetch Interval |  | False |

> | What time field should we filter incidents by? | Default is to filter by received-time, which works well if the folder is an "Inbox". But for a folder emails are dragged into for attention, if we filter by received-time, out-of-order processing of emails means some are ignored. Filtering by modified-time works better for such a scenario. This works best if any modifications \(such as tagging\) happens before moving the email into the folder, such that the move into the folder is the last modification, and triggers Cortex XSOAR to fetch it as an incident. | False |

Reviewer: ```suggestion
| What time field should we filter incidents by? | Default is to filter by received-time, which works well if the folder is an "Inbox". But for a folder emails are dragged into for attention, if we filter by received-time, out-of-order processing of emails means some are ignored. Filtering by modified-time works better for such a scenario. This works best if any modifications \(such as tagging\) happens before moving the email into the folder, such that the move into the folder is the last modification, and triggers Cortex XSOAR to fetch it as an incident. | False |
```

5. Click **Test** to validate the URLs, token, and connection.

## Fetch Incidents

> #### Description

Reviewer: No need for this heading.  Text should be directly under Fetch Incidents heading.

The integration imports email messages from the destination folder in the target mailbox as incidents. If the message contains any attachments, they are uploaded to the War Room as files. If the attachment is an email, Cortex XSOAR fetches information about the attached email and downloads all of its attachments (if there are any) as files.

To use Fetch incidents, configure a new instance and select the `Fetches incidents` option in the instance settings.

**IMPORTANT**:  

> `First fetch timestamp` field is used to determine how much time back to fetch incidents from. The default value is the previous 10 minutes, Meaning: If this is the first time emails are fetched from the destination folder, all emails from 10 minutes prior to the instance configuration and up to the current time will be fetched.

Reviewer: ```suggestion
`First fetch timestamp` field is used to determine how much time back to fetch incidents from. The default value is the previous 10 minutes, Meaning, if this is the first time emails are fetched from the destination folder, all emails from 10 minutes prior to the instance configuration and up to the current time will be fetched.
```

When set to get a long period of time, the `Timeout` field might need to be set to a higher value.

Pay special attention to the following fields in the instance settings:

> `Email Address` – mailbox to fetch incidents from.  

Reviewer: ```suggestion
- `Email Address` – mailbox to fetch incidents from.  
```

> `Name of the folder from which to fetch incidents` – use this field to configure the destination folder from where emails should be fetched. The default is Inbox folder.

Reviewer: ```suggestion
- `Name of the folder from which to fetch incidents` – use this field to configure the destination folder from where emails should be fetched. The default is Inbox folder.
```

#### Permissions

Impersonation rights required. In order to perform actions on the target mailbox of other users, the service account must be part of the _ApplicationImpersonation_ role.

#### Limitations

> If Exchange is configured with an international flavor `Inbox` will be named according to the configured language.

Reviewer: ```suggestion
If Exchange is configured with an international flavor, `Inbox` will be named according to the configured language.
```

## Commands

<details>
<summary><h3 style={{display: 'inline'}}>ews-get-attachment</h3></summary>

> ### ews-get-attachment

Reviewer: I love that the commands are in a drop down list.  However, when you open the drop down, the name of the command is repeated.  Is there a way to get rid of the second instance of the command name?

> #### Description

Reviewer: Do we need the Description heading?  I would prefer if the text was directly under the command name.

Retrieves the actual attachments from an email message. To get all attachments for a message, only specify the item-id argument.

#### Permissions

Impersonation rights required. In order to perform actions on the target mailbox of other users, the service account must be part of the _ApplicationImpersonation_ role.

#### Limitations

No known limitations.

#### Inputs

|**Argument Name**|**Description**|**Required**|
|--- |--- |--- |
|item-id |The ID of the email message for which to get the attachments.|Required|
|target-mailbox |The mailbox in which this attachment was found. If empty, the default mailbox is used. Otherwise, the user might require impersonation rights to this mailbox.|Optional|

> |attachment-ids |The attachments ids to get. If none - all attachments will be retrieved from the message. Support multiple attachments with comma-separated value or array.|Optional|

Reviewer: ```suggestion
|attachment-ids |The attachments IDs to get. If none, all attachments will be retrieved from the message. Support multiple attachments with comma-separated values or an array. |Optional|
```

# EWS O365

## Introduction

Exchange Web Services (EWS) provides the functionality to enable client applications to communicate with the Exchange server. EWS provides access to much of the same data that is made available through Microsoft Office Outlook.

The EWS O365 integration implants EWS leading services. The integration allows getting information on emails and activities in a target mailbox, and some active operations on the mailbox such as deleting emails and attachments or moving emails from folder to folder.

## Supported Versions and license needed

**?**
 
## Use cases

The EWS integration can be used for the following use cases.

* Monitor a specific email account and create incidents from incoming emails to the defined folder.  

>     Follow the instructions in the Fetched Incidents Data section.

Author: Its a few sections after, the Fetch Incidents section? I cannot add a link before having this in production, right? cause the section does not exist.

>     Follow the instructions in the Fetched Incidents Data section.

Reviewer: You should be able to make the link.  It will look something like this:
[Fetch Data](#fetch-data)

for more information on this permission, visit [the Microsoft documentation](https://docs.microsoft.com/en-us/exchange/client-developer/exchange-web-services/how-to-authenticate-an-ews-application-by-using-oauth#configure-for-app-only-authentication).

To limit the application's permissions to only specific mailboxes, follow the [Microsoft documentation](https://docs.microsoft.com/en-us/graph/auth-limit-mailbox-access). Note that it may take about an hour for permissions changes to take effect.

**Note** You can't manage the _Office 365 Exchange Online_ app permissions via the Azure portal.

## Configure Integration on XSOAR (Parameters)

1. Navigate to **Settings** > **Integrations** > **Servers & Services**.
2. Search for EWS O365.
3. Click **Add instance** to create and configure a new integration instance.  

| **Parameter** | **Description** |**Required**|
| --- | --- | --- |
| ID / Application ID | ID can be received after following the System Integration Setup (Device side steps). | False |
| Token / Tenant ID | Token can be received after following the System Integration Setup (Device side steps). | False |

> | Key / Application Secret | Key can be received after following the System Integration Setup (Device side steps). | False |

Author: this title in from 2 sections above.
System Integration Setup (Device side steps)

> | Key / Application Secret | Key can be received after following the System Integration Setup (Device side steps). | False |

Reviewer: I am in discussion with Nadav about that heading. Perhaps just:
Set up the integrated system

| Azure Cloud | Azure Cloud environment. Options are: _Worldwide_ (The publicly accessible Azure Cloud), _US GCC_ (Azure cloud for the USA Government Cloud Community), _US GCC-High_ (Azure cloud for the USA Government Cloud Community High), _DoD_ (Azure cloud for the USA Department of Defense), _Germany_ (Azure cloud for the German Government) | False|

> | Email Address | Mailbox to run commands on and to fetch incidents from. To use this functionality, your account must have impersonation rights or delegation for the account specified. For more information, see https://xsoar.pan.dev/docs/reference/integrations/ewso365\#additional-information | True |

Reviewer 2: the link is no longer exist

Reviewer: I fixed the link

There are two application authentication methods available.
Follow your preferred method's guide on how to use the admin consent flow in order to receive your authentication information:

* [Cortex XSOAR Application](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#cortex-xsoar-application)
* [Self-Deployed Application](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#self-deployed-application) - Client Credential Flow.

## Authentication

For more details about the authentication used in this integration, see [Microsoft Integrations - Authentication](https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication).

## Permissions

In order to function as expected, the service account should have:

**Impersonation rights** - In order to perform actions on the target mailbox of other users, the _service account_ must be part of the `ApplicationImpersonation` role. For more information and instructions on how to set up the permission, see [Microsoft Documentation](https://learn.microsoft.com/en-us/exchange/client-developer/exchange-web-services/impersonation-and-ews-in-exchange).

> Most command require this permission to function correctly. This permission is specified in each relevant command's Permission section. For more information, see [Microsoft Documentation](https://learn.microsoft.com/en-us/exchange/client-developer/exchange-web-services/impersonation-and-ews-in-exchange). 

Reviewer 3: ```suggestion
Most commands require this permission to function correctly. This permission is specified in each relevant command's Permission section. For more information, see [Microsoft Documentation](https://learn.microsoft.com/en-us/exchange/client-developer/exchange-web-services/impersonation-and-ews-in-exchange). 
```

Reviewer 3: also. is this intended: [Microsoft Documentation] ?

Author: Yes it is a link to the docs.

2. Search for your app under *all applications*.
3. Click **API permissions** > **Add permission**.
4. Search for `Office 365 Exchange Online` API > `Application Permission`> `full_access_as_app` permission. 

For more information on this permission, see [the Microsoft documentation](https://docs.microsoft.com/en-us/exchange/client-developer/exchange-web-services/how-to-authenticate-an-ews-application-by-using-oauth#configure-for-app-only-authentication).

To limit the application's permissions to only specific mailboxes, follow the [Microsoft documentation](https://docs.microsoft.com/en-us/graph/auth-limit-mailbox-access). Note that it may take about an hour for permissions changes to take effect.

## Configure Integration on Cortex

| **Parameter** | **Description** |**Required**|
| --- | --- | --- |
| ID / Application ID | ID can be received after following the System Integration Setup (Device side steps). | False |
| Token / Tenant ID | Token can be received after following the System Integration Setup (Device side steps). | False |
| Key / Application Secret | Key can be received after following the System Integration Setup (Device side steps). | False |

> | Azure Cloud | Azure Cloud environment. Options are: _Worldwide_ (The publicly accessible Azure Cloud), _US GCC_ (Azure cloud for the USA Government Cloud Community), _US GCC-High_ (Azure cloud for the USA Government Cloud Community High), _DoD_ (Azure cloud for the USA Department of Defense), _Germany_ (Azure cloud for the German Government) | False|

Reviewer 4: ```suggestion
| Azure Cloud | Azure Cloud environment. Options are: _Worldwide_ (The publicly accessible Azure Cloud), _US GCC_ (Azure cloud for the USA Government Cloud Community), _US GCC-High_ (Azure cloud for the USA Government Cloud Community High), _DoD_ (Azure cloud for the USA Department of Defense), _Germany_ (Azure cloud for the German Government), _China_ (Azure cloud for the Chinese Government ) | False|

```

</details>
<details>
<summary><h3 style={{display: 'inline'}}>ews-delete-attachment</h3></summary>

### ews-delete-attachment

#### Description

Deletes the attachments of an item (email message).

#### Permissions

Impersonation rights required. In order to perform actions on the target mailbox of other users, the service account must be part of the _ApplicationImpersonation_ role.

#### Limitations

No known limitations.

#### Inputs

|**Argument Name**|**Description**|**Required**|
|--- |--- |--- |
|item-id|The ID of the email message for which to delete attachments.|Required|
|target-mailbox|The mailbox in which this attachment was found. If empty, the default mailbox is used. Otherwise, the user might require impersonation rights to this mailbox.|Optional|

> |attachment-ids|A CSV list (or array) of attachment IDs to delete. If empty, all attachments will be deleted from the message.|Optional|

Reviewer: For consistency, we should use "A comma-separated list (or array).....
Update in all places.

            }
        }
    }
```

</details>

<details>
<summary><h3 style={{display: 'inline'}}>ews-delete-items</h3></summary>

### ews-delete-items

#### Description

Delete an item from a mailbox

#### Permissions

Impersonation rights required. In order to perform actions on the target mailbox of other users, the service account must be part of the _ApplicationImpersonation_ role.

#### Limitations

No known limitations.

#### Inputs

|**Argument Name**|**Description**|**Required**|
|--- |--- |--- |

> |item-ids|The item IDs to delete.|Required|

Reviewer: Is this a comma-separated list of .....

> |is-public|Whether the folder is a Public Folder?|Optional|

Reviewer: ```suggestion
|is-public|Whether the folder is a public folder?|Optional|
```

|target-mailbox|The mailbox in which the messages found. If empty, will use the default mailbox. If you specify a different mailbox, you might need impersonation rights to the mailbox.|Optional|

> |is-public|Whether the target folder is a Public Folder.|Optional|

Reviewer: ```suggestion
|is-public|Whether the target folder is a public folder.|Optional|
```

#### Examples

```
!ews-create-folder folder-path=Inbox new-folder-name="Created Folder" target-mailbox=test@demistodev.onmicrosoft.com
```

##### Human Readable Output
>
> Folder Inbox\Created Folder created successfully

</details>

<details>
<summary><h3 style={{display: 'inline'}}>ews-mark-item-as-junk</h3></summary>

### ews-mark-item-as-junk

#### Description

Marks an item as junk. This is commonly used to block an email address. For more information, see the [Microsoft documentation](https://msdn.microsoft.com/en-us/library/office/dn481311(v=exchg.150).aspx).

#### Permissions

Impersonation rights are required. To perform actions on the target mailbox of other users, the service account must be part of the _ApplicationImpersonation_ role.

#### Limitations

No known limitations.

#### Inputs

|**Argument Name**|**Description**|**Required**|
|--- |--- |--- |
|item-id|The item ID to mark as junk.|Required|
|move-items|Whether to move the item from the original folder to the junk folder.|Optional|
|target-mailbox|If empty, will use the default mailbox. If you specify a different mailbox, you might need impersonation rights to the mailbox.|Optional|

#### Outputs

> ```
> There is no context output for this command.
> ```

Reviewer: ```suggestion
There is no context output for this command.
```

```

</details>

<details>
<summary><h3 style={{display: 'inline'}}>ews-find-folders</h3></summary>

### ews-find-folders

#### Description

Retrieves information for the folders of the specified mailbox. Only folders with read permissions will be returned. Your visual folders on the mailbox, such as "Inbox", are under the folder "Top of Information Store".

#### Permissions

Impersonation rights are required. To perform actions on the target mailbox of other users, the service account must be part of the _ApplicationImpersonation_ role.

#### Limitations

No known limitations.

#### Inputs

|**Argument Name**|**Description**|**Required**|
|--- |--- |--- |
|target-mailbox|The mailbox on which to apply the command.|Optional|

> |is-public|Whether to find Public Folders.|Optional|

Reviewer: ```suggestion
|is-public|Whether to find public folders.|Optional|
```

```

</details>

<details>
<summary><h3 style={{display: 'inline'}}>ews-get-items-from-folder</h3></summary>

### ews-get-items-from-folder

#### Description

> Retrieves items from a specified folder in a mailbox. The items are ordered by the item created time, most recent is first.

Reviewer: ```suggestion
Retrieves items from a specified folder in a mailbox. The items are ordered by the item created time. Most recent is first.
```

> |is-public|Whether the folder is a Public Folder. Default is 'False'.|Optional|

Reviewer: ```suggestion
|is-public|Whether the folder is a public folder. Default is 'False'.|Optional|
```

```

</details>

<details>
<summary><h3 style={{display: 'inline'}}>ews-get-folder</h3></summary>

### ews-get-folder

#### Description

Retrieves a single folder.

#### Permissions

Impersonation rights are required. To perform actions on the target mailbox of other users, the service account must be part of the _ApplicationImpersonation_ role.

#### Limitations

> If Exchange is configured with an international flavor `Inbox` will be named according to the configured language.

Reviewer: ```suggestion
If Exchange is configured with an international flavor, `Inbox` will be named according to the configured language.
```

#### Inputs

|**Argument Name**|**Description**|**Required**|
|--- |--- |--- |
|target-mailbox|The mailbox on which to apply the search.|Optional|
|folder-path|The path of the folder to retrieve. If empty, will retrieve the folder "AllItems".|Optional|

> |is-public|Whether the folder is a Public Folder. Default is "False".|Optional|

Reviewer: ```suggestion
|is-public|Whether the folder is a public folder. Default is "False".|Optional|
```

```

</details>

<details><summary><h3 style={{display: 'inline'}}>send-mail</h3></summary>

### Description
Sends an email.

### Permissions
Impersonation rights are required. To perform actions on the target mailbox of other users, the service account must be part of the ApplicationImpersonation role.

### Limitations

> No known limitations.

Author: ```suggestion
When sending the email to an Outlook account, Outlook UI fails to display custom headers. This does not happens when sending to Gmail account.
```

```

</details>

<details>
<summary><h3 style={{display: 'inline'}}>send-mail</h3></summary>

### send-mail

#### Description

Sends an email.

#### Permissions

Impersonation rights are required. To perform actions on the target mailbox of other users, the service account must be part of the _ApplicationImpersonation_ role.

#### Limitations

> When sending the email to an Outlook account, Outlook UI fails to display custom headers. This does not happens when sending to Gmail account.

Reviewer: ```suggestion
When sending the email to an Outlook account, Outlook UI fails to display custom headers. This does not happen when sending to a Gmail account.
```

#### Outputs

There is no context output for this command.

#### Examples

```!reply-mail item_id=AAMkAGY3OTQyMzMzLWYxNjktNDE0My05NmZhLWQ5MGY1YjIyNzBkNABGAAAAAACYCKjWAnXBTrnhgWJCcLX7BwDrxRwRjq/zTrN6vWSzK4OWAAAAAAEMAADrxRwRjq/zTrN6vWSzK4OWAAPYQGFeAAA= body=hello subject=hi to="avishai@demistodev.onmicrosoft.com"```

##### Human Readable Output

##### Sent email

>|attachments|from|subject|to|
>|---|---|---|---|
>|  | avishai@demistodev.onmicrosoft.com | hi | avishai@demistodev.onmicrosoft.com |

</details>

<details>
<summary><h3 style={{display: 'inline'}}>ews-auth-reset</h3></summary>

### ews-auth-reset

Run this command if for some reason you need to rerun the authentication process.

#### Permissions

No additional permissions are needed.

#### Limitations

No known limitations.

#### Inputs

There is no input for this command.

#### Outputs

There is no context output for this command.

</details>

## Troubleshooting

<details><summary><h3 style={{display: 'inline'}}>Instance Configuration </h3></summary> No troubleshooting found. </details>

<details><summary><h3 style={{display: 'inline'}}> Fetch command </h3></summary>

> * If incidents are not being fetched, verify no `pre-process` rule is configured that might filter some incidents out.

Reviewer: ```suggestion
* If incidents are not being fetched, verify that no `pre-process` rule is configured that might filter some incidents out.
```

> * "address parts cannot contain CR or LF" error message in the logs means an corrupted email might have failed the process. In order to solve, you might need to remove the email from the folder being fetched. Contact Support Team if you believe the email is not corrupted. 

Reviewer: ```suggestion
* "address parts cannot contain CR or LF" error message in the logs means a corrupted email might have failed the process. In order to resolve this, you might need to remove the email from the folder being fetched. Contact Support Team if you believe the email is not corrupted. 
```

</details>

<details><summary><h3 style={{display: 'inline'}}> General </h3></summary> 

* ews-get-searchable-mailboxes:   

> When using UPN parameter, the command ews-get-searchable-mailboxes would work after assigning RBAC roles requested in the management role header as explained in the [Microsoft Documentation](https://learn.microsoft.com/en-us/Exchange/policy-and-compliance/ediscovery/assign-permissions?redirectedfrom=MSDN&view=exchserver-2019).

Reviewer: ```suggestion
When using the UPN parameter, the command ews-get-searchable-mailboxes runs correctly after assigning RBAC roles requested in the management role header as explained in the [Microsoft Documentation](https://learn.microsoft.com/en-us/Exchange/policy-and-compliance/ediscovery/assign-permissions?redirectedfrom=MSDN&view=exchserver-2019).
```

| attachCIDs | A comma-separated list of CIDs to embed attachments within the email itself. | Optional | 

#### Outputs

There is no context output for this command.

#### Examples

```!reply-mail item_id=AAMkAGY3OTQyMzMzLWYxNjktNDE0My05NmZhLWQ5MGY1YjIyNzBkNABGAAAAAACYCKjWAnXBTrnhgWJCcLX7BwDrxRwRjq/zTrN6vWSzK4OWAAAAAAEMAADrxRwRjq/zTrN6vWSzK4OWAAPYQGFeAAA= body=hello subject=hi to="avishai@demistodev.onmicrosoft.com"```

##### Human Readable Output

##### Sent email

>|attachments|from|subject|to|
>|---|---|---|---|
>|  | avishai@demistodev.onmicrosoft.com | hi | avishai@demistodev.onmicrosoft.com |

</details>

<details>
<summary><h3 style={{display: 'inline'}}>ews-auth-reset</h3></summary>

### ews-auth-reset

#### Description

Rerun the authentication process.

#### Permissions

> No additional Permissions needed.

Reviewer: ```suggestion
No additional permissions are needed.
```

#### Limitations

No known limitations.

#### Inputs

There is no input for this command.

#### Outputs

There is no context output for this command.

#### Examples

##### Human Readable Output

>#### Sent email

>|attachments|from|subject|to|
>|---|---|---|---|
>|  | avishai@demistodev.onmicrosoft.com | hi | avishai@demistodev.onmicrosoft.com |

</details>

## Troubleshooting

<details><summary><h3 style={{display: 'inline'}}>Instance Configuration </h3></summary> No troubleshooting found. </details>

<details><summary><h3 style={{display: 'inline'}}> Fetch command </h3></summary>

* If incidents are not being fetched, verify no `pre-process` rule is configured that might filter some incidents out.

> * "address parts cannot contain CR or LF" error message in the logs means an corrupted email might have fail the process. In order to solve you might need to remove the email from the folder being fetched. Contact Support Team if you believe the email is not corrupted. 

Reviewer: ```suggestion
* "address parts cannot contain CR or LF" error message in the logs means an corrupted email might have failed the process. In order to solve, you might need to remove the email from the folder being fetched. Contact Support Team if you believe the email is not corrupted. 
```

| attachCIDs | A comma-separated list of CIDs to embed attachments within the email itself. | Optional | 

#### Outputs

There is no context output for this command.

#### Examples

```!reply-mail item_id=AAMkAGY3OTQyMzMzLWYxNjktNDE0My05NmZhLWQ5MGY1YjIyNzBkNABGAAAAAACYCKjWAnXBTrnhgWJCcLX7BwDrxRwRjq/zTrN6vWSzK4OWAAAAAAEMAADrxRwRjq/zTrN6vWSzK4OWAAPYQGFeAAA= body=hello subject=hi to="avishai@demistodev.onmicrosoft.com"```

##### Human Readable Output

##### Sent email

>|attachments|from|subject|to|
>|---|---|---|---|
>|  | avishai@demistodev.onmicrosoft.com | hi | avishai@demistodev.onmicrosoft.com |

</details>

<details>
<summary><h3 style={{display: 'inline'}}>ews-auth-reset</h3></summary>

### ews-auth-reset

#### Description

Rerun the authentication process.

#### Permissions

No additional Permissions needed.

#### Limitations

No known limitations.

#### Inputs

There is no input for this command.

#### Outputs

There is no context output for this command.

> #### Examples

Reviewer 2: the example is not correct
